pkg/logs/sds/scanner.go

```diff
@@ -225,28 +225,50 @@ func interpretRCRule(userRule RuleConfig, standardRule StandardRuleConfig) (sds.
 		extraConfig.ProximityKeywords = sds.CreateProximityKeywordsConfig(userRule.IncludedKeywords.CharacterCount, userRule.IncludedKeywords.Keywords, nil)
 	}
 
-	// the RC schema supports multiple of them,
-	// for now though, the lib only supports one, so we'll just use the first one.
-	if len(standardRule.SecondaryValidators) > 0 {
-		received := standardRule.SecondaryValidators[0]
-		switch received.Type {
-		case RCSecondaryValidationChineseIdChecksum:
-			extraConfig.SecondaryValidator = sds.ChineseIdChecksum
-		case RCSecondaryValidationLuhnChecksum:
-			extraConfig.SecondaryValidator = sds.LuhnChecksum
-		default:
-			log.Warnf("Unknown secondary validator: ", string(received.Type))
-			// TODO(remy): telemetry
+	var defToUse = StandardRuleDefinition{Version: -1}
+
+	// go through all received definitions, use the most recent supported one.
+	// O(n) number of definitions in the rule.
+	for _, stdRuleDef := range standardRule.Definitions {
+		if defToUse.Version > stdRuleDef.Version {
+			continue
 		}
+
+		// The RC schema supports multiple of them,
+		// for now though, the lib only supports one, so we'll just use the first one.
+		if len(stdRuleDef.SecondaryValidators) > 0 {
+			received := stdRuleDef.SecondaryValidators[0]
+			switch received.Type {
+			case RCSecondaryValidationChineseIdChecksum:
+				extraConfig.SecondaryValidator = sds.ChineseIdChecksum
+				defToUse = stdRuleDef
+			case RCSecondaryValidationLuhnChecksum:
+				extraConfig.SecondaryValidator = sds.LuhnChecksum
+				defToUse = stdRuleDef
+			default:
+				// we don't know this secondary validator, test another version
+				log.Warnf("unknown secondary validator: ", string(received.Type))
```

Comment:
Would we only see this warning if the user is actually enabling a standard rule with secondary validation that we don't recognize rather than if there are _any_ standard rules (even ones the user hasn't enabled)?

Reply:
Oh I think I see what you mean: it's only when there is an user rule enabling the usage of a standard rule (with unknown secondary validators), i.e., it's not just when receiving standard rules.

Reply:
Makes sense. I just wanted to confirm since otherwise the warnings might be spurious.